src/query/storages/stage/src/stage_table.rs

```diff
@@ -73,17 +74,17 @@ impl StageTable {
     }
 
     /// Get operator with correctly prefix.
-    pub fn get_op(ctx: &Arc<dyn TableContext>, stage: &UserStageInfo) -> Result<Operator> {
+    pub fn get_op(dop: &DataOperator, stage: &UserStageInfo) -> Result<Operator> {
         if stage.stage_type == StageType::External {
             Ok(init_operator(&stage.stage_params.storage)?)
         } else {
-            let pop = ctx.get_data_operator()?.operator();
+            let pop = dop.operator();
```

Comment:
Maybe we can use `DataOperator::instance()` here directly?

Reply:
Is it OK to build a `DataOperator::instance()` for every operation?

Reply:
hmmm, it seems `get_data_operator` also cloned the shared `DataOperator` every time.

Reply:
Don't worry, DataOperator is cheap to clone. We have an internal `Arc`.